src/zarr/storage/_cache.py

```diff
@@ -0,0 +1,735 @@
+                self.hits += 1
+                # treat the end as most recently used
+                self._values_cache.move_to_end(cache_key)
+
+        except KeyError:
+            # cache miss, retrieve value from the store
+            if hasattr(self._store, "__getitem__"):
+                value = self._store[key]
+            else:
+                # Fallback for async stores
+                raise KeyError(f"Key {key} not found in store") from None
+            with self._mutex:
+                self.misses += 1
+                # need to check if key is not in the cache, as it may have been cached
+                # while we were retrieving the value from the store
+                if cache_key not in self._values_cache:
+                    self._cache_value(cache_key, value)
+
+        return value
+
+    def __setitem__(self, key: str, value: Buffer) -> None:
+        if hasattr(self._store, "__setitem__"):
+            self._store[key] = value
+        else:
+            # For async stores, we can't handle this synchronously
+            raise TypeError("Cannot use __setitem__ with async store")
+
+        # Update cache and invalidate keys cache since we may have added a new key
+        with self._mutex:
+            self._invalidate_keys()
+            self._cache_value(self._normalize_key(key), value)
+
+    def __delitem__(self, key: Any) -> None:
+        if hasattr(self._store, "__delitem__"):
+            del self._store[key]
+        else:
+            # For async stores, this shouldn't be used - use delete() instead
+            raise NotImplementedError("Use async delete() method for async stores")
+        with self._mutex:
+            self._invalidate_keys()
+            cache_key = self._normalize_key(key)
+            self._invalidate_value(cache_key)
+
+    def __eq__(self, value: object) -> bool:
+        return type(self) is type(value) and self._store.__eq__(value._store)  # type: ignore[attr-defined]
+
+    async def delete(self, key: str) -> None:
+        """
+        Remove a key from the store.
+
+        Parameters
+        ----------
+        key : str
+
+        Notes
+        -----
+        If ``key`` is a directory within this store, the entire directory
+        at ``store.root / key`` is deleted.
+        """
+        # Check if store is writable
+        self._check_writable()
+
+        # Delegate to the underlying store for actual deletion
+        if hasattr(self._store, "delete"):
+            await self._store.delete(key)
+        else:
+            # Fallback for stores that don't have async delete
+            del self._store[key]  # type: ignore[attr-defined]
+
+        # Invalidate cache entries
+        with self._mutex:
+            self._invalidate_keys()
+            cache_key = self._normalize_key(key)
+            self._invalidate_value(cache_key)
+
+    async def exists(self, key: str) -> bool:
+        # Delegate to the underlying store
+        if hasattr(self._store, "exists"):
```

Comment:
All stores define `exists()` (it's an abstract method on `Store`), so no need for this logic.